browser/psst/psst_tab_web_contents_observer_browsertest.cc

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/psst/psst_tab_web_contents_observer.h"
+
+#include <memory>
+
+#include "base/containers/contains.h"
+#include "base/path_service.h"
+#include "base/test/scoped_feature_list.h"
+#include "brave/browser/ui/brave_browser.h"
+#include "brave/browser/ui/tabs/public/tab_features.h"
+#include "brave/components/constants/brave_paths.h"
+#include "brave/components/psst/browser/content/psst_scripts_result_handler.h"
+#include "brave/components/psst/browser/core/psst_rule_registry.h"
+#include "brave/components/psst/buildflags/buildflags.h"
+#include "brave/components/psst/common/features.h"
+#include "brave/components/psst/common/pref_names.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/test/base/chrome_test_utils.h"
+#include "chrome/test/base/platform_browser_test.h"
+#include "chrome/test/base/testing_browser_process.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "content/public/test/content_mock_cert_verifier.h"
+#include "net/dns/mock_host_resolver.h"
+#include "net/test/embedded_test_server/embedded_test_server.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace psst {
+
+class MockScriptHandler : public PsstScriptsHandler {
+ public:
+  MockScriptHandler() = default;
```

Comment:
you shouldn't be mocking the script handler for a browser test, that is also mocking the system under test. This should only be done for unit tests